webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue

```diff
@@ -458,73 +451,11 @@ export default {
-                          </i>
-                          <del>${profile.fullname}${profile.isExternal === 'true' ? ` (${  this.$t('UsersManagement.type.external')  })` : ''}</del>
-                          <a href="#" class="remove">
-                            <i class="uiIconClose uiIconLightGray"></i>
-                          </a>
-                        </span>
-                      </span>
-                    `));
-    },
     resetRichEditorData() {
-      let message = this.ckEditorInstance.getMessage();
+      const message = this.ckEditorInstance.getMessage();
       const mentionedUsers =  message.match(/@([A-Za-z0-9_'.+-]+)/g)?.map(a => a.replace('@', '')) || null;
-      this.containInvalidUsers = false;
       if (mentionedUsers?.length) {
-        // Clear suggester cache
-        document.dispatchEvent(new CustomEvent('suggester-clear-people-cache'));
-        Promise
-          .all(mentionedUsers.map(username => {
-            return this.$identityService.getIdentityByProviderIdAndRemoteId('organization', username)
-              .then(identity => {
-                if (!identity?.profile) {
-                  return null;
-                }
-                if (this.audience?.spaceId) {
-                  return this.$spaceService.isSpaceMember(this.audience.spaceId, username)
-                    .then(data => {
-                      if (data) {
-                        const profile = identity.profile;
-                        profile.isMember = data?.isMember === 'true';
-                        return profile;
-                      } else {
-                        return null;
-                      }
-                    });
-                } else {
-                  const profile = identity.profile;
-                  profile.isMember = true;
-                  return profile;
-                }
-              });
-          }))
-          .then(userProfiles => userProfiles.filter(p => p))
-          .then(userProfiles => {
-            const containsExoMentionClass = message.search('exo-mention') >= 0;
-            this.containInvalidUsers = !!userProfiles.find(profile => profile.isMember !== true);
-            userProfiles.forEach(profile => {
-              const pattern = containsExoMentionClass ? `<span class="atwho-inserted" data-atwho-at-query="@${profile.username}"[^>]*>(.*?)</span> </span>` : `@${profile.username}`;
-              if (profile.isMember) {
-                message = this.replaceValidSuggestedUser(message, profile, pattern);
-              } else {
-                message = this.replaceInvalidSuggestedUser(message, profile, pattern);
-              }
-            });
-            this.ckEditorInstance?.initCKEditorData?.(message);
-          });
+        this.ckEditorInstance?.replaceSuggestedUsers?.(message, mentionedUsers, this.spaceId);
```

Comment:
I think it Would be better to make this call in method `getContentToEdit` of `RichEditor.vue`
```suggestion
```

Reply:
done!